src/violations.rs

```diff
@@ -48,6 +48,16 @@ impl Violation for LineTooLong {
     }
 }
 
+define_violation!(
+    pub struct MixedSpacesAndTabs;
+);
+impl Violation for MixedSpacesAndTabs {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Indentation contains mixed spaces and tabs")
+    }
+}
```

Comment:
(We're moving towards a model in which every rule gets its own file (rather than a `rules.rs`), and every `Violation` is defined in that file rather than here. But, the `pycodestyle` rules don't follow that convention yet, and I'd rather be consistent with a rule set, as opposed to doing this one rule in the new style. So, no changes needed, just adding a comment for... well, IDK why, for any onlookers I guess.)

Reply:
I have a bunch more pycodestyle things I'd like to add, and I don't want to keep adding to a direction you'd like to change. How could I help make this change to rule files happen? Are there examples I could look at so I could attempt the refactoring?

Reply:
Yeah it's mostly a mechanical thing. Let me pull up an example...

Reply:
This is an example of breaking up into one rule per file: https://github.com/charliermarsh/ruff/pull/2003/files.

Reply:
This is an example of placing the violation directly in the rule file (rather than in `violations.rs`): https://github.com/charliermarsh/ruff/pull/2023/files#diff-535e9b92bc3115ba9761fc71b63b3a70c16865330ccd3a080cd6eabdf8013faeR13

Reply:
Let me know if you have any questions, I'm around :)